sw/host/opentitanlib/src/transport/mod.rs

```diff
@@ -88,12 +91,21 @@ pub trait Transport {
     fn gpio_pin(&self, _instance: &str) -> Result<Rc<dyn GpioPin>> {
         unimplemented!();
     }
-    /// Programs a bitstream into an FPGA.
-    fn fpga_program(&self, _bitstream: &[u8]) -> Result<()> {
-        unimplemented!();
+    /// Invoke non-standard functionality of some Transport implementations.
+    fn dispatch(&self, _action: &mut dyn Any) -> Result<Option<Box<dyn Serialize>>> {
+        Err(TransportError::UnsupportedOperation.into())
     }
 }
 
+/// Used by Transport implementations dealing with emulated OpenTitan
+/// chips, allowing e.g. more efficient direct means of programming
+/// emulated flash storage.  (As opposed to running an actual
+/// bootloater on the emulated target, which would receive data via
+/// SPI to be flashed.)
+pub struct Bootstrap {
+    pub image_path: Box<std::path::Path>,
+}
```

Comment:
If you use PathBuf here, you shouldn't have to `Box` it.  Path is to PathBuf as &str is to String.

Yes, you'll still have to `clone` the original PathBuf to make a copy, but that's no worse than a strdup for this case.

Reply:
Done